starky/src/lib.rs

```diff
@@ -1,14 +1,333 @@
+//!         &self,
+//!         vars: &Self::EvaluationFrame<FE, P, D2>,
+//!         yield_constr: &mut ConstraintConsumer<P>,
+//!     ) where
+//!         FE: FieldExtension<D2, BaseField = F>,
+//!         P: PackedField<Scalar = FE>,
+//!     {
+//!         let local_values = vars.get_local_values();
+//!         let next_values = vars.get_next_values();
+//!         let public_inputs = vars.get_public_inputs();
+//!
+//!         // Check public inputs.
+//!         yield_constr.constraint_first_row(local_values[0] - public_inputs[Self::PI_INDEX_X0]);
+//!         yield_constr.constraint_first_row(local_values[1] - public_inputs[Self::PI_INDEX_X1]);
+//!         yield_constr.constraint_last_row(local_values[1] - public_inputs[Self::PI_INDEX_RES]);
+//!
+//!         // Enforce the Fibonacci transition constraints.
+//!         // x0' <- x1
+//!         yield_constr.constraint_transition(next_values[0] - local_values[1]);
+//!         // x1' <- x0 + x1
+//!         yield_constr.constraint_transition(next_values[1] - local_values[0] - local_values[1]);
+//!     }
+//!
+//!     // Define the constraints to recursively verify this STARK.
+//!     fn eval_ext_circuit(
+//!         &self,
+//!         builder: &mut CircuitBuilder<F, D>,
+//!         vars: &Self::EvaluationFrameTarget,
+//!         yield_constr: &mut RecursiveConstraintConsumer<F, D>,
+//!     ) {
+//!         let local_values = vars.get_local_values();
+//!         let next_values = vars.get_next_values();
+//!         let public_inputs = vars.get_public_inputs();
+//!
+//!         // Check public inputs.
+//!         let pis_constraints = [
+//!             builder.sub_extension(local_values[0], public_inputs[Self::PI_INDEX_X0]),
+//!             builder.sub_extension(local_values[1], public_inputs[Self::PI_INDEX_X1]),
+//!             builder.sub_extension(local_values[1], public_inputs[Self::PI_INDEX_RES]),
+//!         ];
+//!
+//!         yield_constr.constraint_first_row(builder, pis_constraints[0]);
+//!         yield_constr.constraint_first_row(builder, pis_constraints[1]);
+//!         yield_constr.constraint_last_row(builder, pis_constraints[2]);
+//!
+//!         // Enforce the Fibonacci transition constraints.
+//!         // x0' <- x1
+//!         let first_col_constraint = builder.sub_extension(next_values[0], local_values[1]);
+//!         yield_constr.constraint_transition(builder, first_col_constraint);
+//!         // x1' <- x0 + x1
+//!         let second_col_constraint = {
+//!             let tmp = builder.sub_extension(next_values[1], local_values[0]);
+//!             builder.sub_extension(tmp, local_values[1])
+//!         };
+//!         yield_constr.constraint_transition(builder, second_col_constraint);
+//!     }
+//!
+//!     fn constraint_degree(&self) -> usize {
+//!         2
+//!     }
+//! }
+//! ```
+//!
+//! One can then instantiate a new `FibonacciStark` instance, generate an associated
+//! STARK trace, and generate a proof for it.
+//!
```

Comment:
I think the witness generation and constraints are written twice in this file (and should therefore be removed from below). What follows should start at `fn fibonacci<F: Field>(n: usize, x0: F, x1: F) -> F {`

Reply:
It wouldn't compile if this wasn't included. Rustdoc starting with ` # ` does not print the line but uses it for compiling / running doctest. You can see it doesn't appear twice by calling `cargo doc -p starky --no-deps --open`.